### ISSUE TITLE
(Feature): support GPU encoding (qsv nvenc amf)

### ISSUE THREAD

**Issue by shithappens2016** [NatronGitHub/Natron#858]
### Make sure to follow our issue report guidelines

- [X] I have read the [feature request guidelines](https://hackmd.io/@natron-dev-awesome/B1SW6Hbau)
- [X] I have followed the [contributing guidelines](https://github.com/NatronGitHub/Natron/blob/RB-2.4/CONTRIBUTING.md) to the best of my understanding
- [X] This feature is not on the [issue tracker](https://github.com/NatronGitHub/Natron/issues?q=is%3Abug+type%3Afeature+type%3Aidea) and is not a duplicate

### Provide a description of your feature request

Please support GPU encoding like `h264_qsv` `h264_nvenc` `h264_amf` `vp9_qsv` `hevc_amf` `hevc_nvenc` `hevc_qsv`.

https://trac.ffmpeg.org/wiki/Hardware/QuickSync

https://trac.ffmpeg.org/wiki/HWAccelIntro

### Is this a realistically implementable feature?

- [ ] My feature is realistically implementable
- [ ] My feature is NOT realistically implementable

### Can you contribute in creating this feature?

- [ ] I am unable, or not willing to contribute to creating this feature
- [ ] I am able and willing to contribute to the creation of this feature

**Comment by shithappens2016** [comment]
@rodlie

GPU-accelerated video encoding on a low-power cpu is really necessary

**Comment by rodlie** [comment]
I'm not sure GPU encoding in ffmpeg will improve the rendering that much (if at all). The majority of rendering time happens before you write anything, and that's all CPU.

**Comment by shithappens2016** [comment]
@rodlie 

I was shocked by your answer. You as a graphics software developer, are not aware of the huge acceleration of video encoding by hardware acceleration in ffmpeg.

**Comment by rodlie** [comment]
Yes, but the writer (ffmpeg) is a small part of the rendering process, the actual pixels are processed on the CPU. That's why I doubt it gives any real gains (I can be wrong). I also don't recommend writing video files (except for prores) with Natron.

**Comment by shithappens2016** [comment]
@rodlie

Dude, you should know that uncompressed video frame sequence files take up a huge amount of hard drive space. Not everyone has a large-capacity hard drive.

Some of us, only have cheap laptops. No powerful processor and no large capacity hard drive. But basically an integrated GPU can accelerate video encoding. We just do some simple effects on the video, and then hope that we can quickly compress it into a delicate and small mp4 file.

**Comment by Randrianasulu** [comment]
if Natron can write yuv4mpeg stream to fifo  file (ffmpeg/mjpeg) video format - you probably can just set Natron to render and start ffmpeg with -i argument and suitable hw compressor :)

**Comment by Randrianasulu** [comment]
also, isn't this duplicate of #553, and related to #161?

**Comment by shithappens2016** [comment]
@Randrianasulu

Does Natron support writing to named pipes?
